### PR TITLE
update version to 0.17.0

### PR DESCRIPTION
**Pull request by wash2**

### CODE REVIEW

**Comment by vberger** [comment]
I haven't followed closely enough the recent advances of SCTK, so if you and @i509VCB believe we are at a good place to make a new release, I'll be glad to publish it.

However, I think it'd be better to update the CHANGELOG for this new version. Probably not a detailed list of changes, but at least a general description of what changed since 0.16.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,19 @@
...
 
+## 0.17.0 - 2023-03028
+
+#### Breaking Changes
+
+- `wayland-rs` dependencies are updated to 0.30 and all APIs have changed significantly as a result
+
+#### Additions
+
+- `xkbcommon` is now optional and gated by the `xkbcommon` feature
```
Comment (i509VCB) [inline CHANGELOG.md:13]: I recall 0.16 having libxkbcommon be dlopened. The `xkbcommon` crate does not do so it's a required dependency if the feature is enabled

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,19 @@
...
+- pointer constraints protocol support
+- relative pointer protocol supports
```
Comment (i509VCB) [inline CHANGELOG.md:15]:
```suggestion
- relative pointer protocol support
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,19 @@
...
+## 0.17.0 - 2023-03028
+
+#### Breaking Changes
+
+- `wayland-rs` dependencies are updated to 0.30 and all APIs have changed significantly as a result
+
+#### Additions
+
+- `xkbcommon` is now optional and gated by the `xkbcommon` feature
+- pointer constraints protocol support
```
Comment (i509VCB) [inline CHANGELOG.md:14]: Probably best to name the protocol, `relative-pointer-unstable-v1`.

File: CHANGELOG.md (hunk 2 above)
Comment (i509VCB) [reply CHANGELOG.md:15]: Same here for mentioning the protocol name

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,19 @@
 
 ## Unreleased
 
+## 0.17.0 - 2023-03028
+
+#### Breaking Changes
+
+- `wayland-rs` dependencies are updated to 0.30 and all APIs have changed significantly as a result
+
+#### Additions
```
Comment (i509VCB) [inline CHANGELOG.md:11]: Layer shell support was a new addition, I think that is worth mentioning.

Comment (i509VCB) [inline CHANGELOG.md:11]: New SHM pool types were added.

**Comment by i509VCB** [comment]
> I haven't followed closely enough the recent advances of SCTK, so if you and @i509VCB believe we are at a good place to make a new release, I'll be glad to publish it.
> 
> However, I think it'd be better to update the CHANGELOG for this new version. Probably not a detailed list of changes, but at least a general description of what changed since 0.16.

Documentation is a little lacking but that isn't necessarily a requirement to release (as it could be fixed later). But I wouldn't be surprised if we forget that in the future. Crate level docs at least to give some idea how the library is structured and used. We did suppress the `missing_docs` lint so there is probably a lot of yellow.

I'd be fine merging this pull request with the changes to the changelog, write a small documentation pull request and then I feel a 0.17 release would be ready.

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,21 @@
 
 ## Unreleased
 
+## 0.17.0 - 2023-03028
```
Comment (i509VCB) [inline CHANGELOG.md:5]: Dash needed between the month and day

Previous entries do that

**Review by elinorbgr — APPROVED** [review]

**Pull request merged**